src/http/request.cr

```diff
@@ -95,11 +95,14 @@ class HTTP::Request
     return BadRequest.new unless parts.size == 3
 
     method, resource, http_version = parts
-    HTTP.parse_headers_and_body(io) do |headers, body|
-      return new method, resource, headers, body, http_version
+    begin
+      HTTP.parse_headers_and_body(io) do |headers, body|
+        return new method, resource, headers, body, http_version
```

Comment:
I'm unsure that you need to keep this code organization with added rescue block.

Why not something like

```crystal
  begin
    HTTP.parse_headers_and_body(io) do |headers, body|
      new method, resource, headers, body, http_version
    end
  rescue ex : ArgumentError
    BadRequest.new
  end
```

Reply:
@akzhan that doesn't work, there's a case where `parse_headers_and_body` returns without yielding which isnt handled.